### PR TITLE
Default state for map actions + other minor testing-related fixes

### PR DESCRIPTION
**Pull request by elijahbenizzy**
## Changes

Default state for `MapActions` is now the state that was provided. Lowers the amount needed for a common use-case.

Also did a bit of improvements on the testing side as well -- removed warnings from test suite!

----

> [!IMPORTANT]
> Set default state for `MapActions` and improve test clarity by renaming classes and adding tests.
> 
>   - **Behavior**:
>     - Default state for `MapActions` now uses the provided state, reducing setup for common use-cases in `parallelism.py`.
>   - **Testing**:
>     - Added `test_map_actions_default_state()` in `test_parallelism.py` to verify default state behavior.
>     - Renamed test classes in `test_application.py` for clarity: `TestActionWithoutContext` to `ActionWithoutContext`, `TestActionWithContext` to `ActionWIthContext`, `TestActionWithKwargs` to `ActionWithKwargs`, and `TestActionWithContextTracer` to `ActionWithContextTracer`.
>   - **Misc**:
>     - Replaced `parse_obj` with `model_validate` in `schema.py` for `BeginEntryModel`, `EndEntryModel`, `BeginSpanModel`, `EndSpanModel`, `AttributeModel`, and streaming events.
> 
> <sup>This description was created by </sup>[Ellipsis](https://www.ellipsis.dev?ref=DAGWorks-Inc%2Fburr&utm_source=github&utm_medium=referral)<sup> for 10705c426557ccbea1186bc39375556425dacf02. It will automatically update as commits are pushed.</sup>

### CODE REVIEW

File: tests/integrations/test_burr_pydantic.py [hunk 1]
```diff
@@ -110,8 +110,7 @@ class MyModelWithConfig(pydantic.BaseModel):
         foo: int
         arbitrary: Arbitrary
 
-        class Config:
-            arbitrary_types_allowed = True
+        model_config = ConfigDict(arbitrary_types_allowed=True)
```
Comment (skrawcz) [inline tests/integrations/test_burr_pydantic.py:113]: no pydantic version assumption to make?

**Review by skrawcz — APPROVED** [review]
might need pydantic > 1 pin?

File: tests/integrations/test_burr_pydantic.py (hunk 1 above)
Comment (elijahbenizzy) [reply tests/integrations/test_burr_pydantic.py:113]: Nope, this is available on the later 1.X versions. 0.X won't work but. I think that's OK.

Comment (elijahbenizzy) [reply tests/integrations/test_burr_pydantic.py:113]: `model_validate` is though so I'll add

**Pull request merged**